### DISCUSSION TITLE
Max Files per Upload does not work

### DISCUSSION THREAD

**Discussion by farzadpanahi** [requarks/wiki#6440] [Error / Bug Report]
Wiki.js ver.: 2.5.298

I have changed the Max Files per Upload under Security from 10 to 100:
![image](https://github.com/requarks/wiki/assets/3793563/795f1d88-f10c-4255-892b-3c5017f3cc4e)

When I go to page assets to upload files, it still reports 10 max files:
![image](https://github.com/requarks/wiki/assets/3793563/715324e9-183a-438f-9e4b-910e312c9e87)

I can upload 10 files:
![image](https://github.com/requarks/wiki/assets/3793563/1a225ff9-a6f5-46e0-bab7-869f845d4703)

If I try to select 11 files, it will not load the files to the assets box and won't give me any error: 
![image](https://github.com/requarks/wiki/assets/3793563/efa5fa2d-0f68-4b31-aaf7-2e4d2e743eef)

No error is showing up in the logs either.

**Comment by ghost** [comment]
I have a similar problem. Maybe it is a similar cause? I have the following situation: Changes to the Max Upload Size do not take effect (so it seems). I have a page that is very long with a lot of images and at a certain amount of images inside the article, I get an error message that claims the upload size limit is reached. The error keeps appearing even after raising the Max Upload Size. 

Could those two problems have the same cause? Does wiki.js fail to communicate the changes to a lower layer? to docker or ubuntu?

@farzadpanahi I just reproduced your problem. I am unable to upload 11 or more files at the same time. Could you also try to reproduce my problem? Create a page and copy paste a lot of images in there. maybe 30 or 40, so that the file size is big. And then try to save the page. A network error should occur that does not let you save. Would be great to know if it also happens for you and see if those 2 issues have the same origin.

I got wiki.js version 2.5.294 btw

**Comment by fvogl** [comment]
Same for me. Can't upload more than 10 files although I've increased the limit.

![image](https://github.com/requarks/wiki/assets/26383910/c84e7517-249d-4cfd-9d35-f3ef7c5add85)

**Comment by SgtErnestBilko** [comment]
I have wikijs v2.5.301

I have changed max file size to 10485760 in both yml file and under security in admistration.  I have restarted wikijs (it is in a docker container), but still my max file size if 5mb.

Has anyone found a solution to this or is it a bug ?

**Reply by phokz** [reply]
Hi,
I had similar issue with maximum file size, found out that instances with mysql database had this issue, but one instance I had on postgresql did work.

Here is my migration script https://github.com/phokz/wikijs-mysql-to-postgres/

Here is my nginx reverse proxy config file
```
server {
    listen       443 ssl;
    listen       [::]:443 ssl;
    server_name  wiki.domain.tld;
    proxy_buffering off;

    client_max_body_size 50m;

    ssl_certificate      /root/.acme.sh/wiki.domain.tld/fullchain.cer;
    ssl_certificate_key  /root/.acme.sh/wiki.domain.tld/wiki.domain.tld.key;

    include /etc/nginx/ssl-options.conf;

    access_log /var/log/nginx/wiki-access.log  main;
    error_log  /var/log/nginx/wiki-error.log;

    location / {
      resolver 8.8.8.8;
      proxy_pass http://localhost:4000/;
      proxy_set_header    X-Real-IP       $remote_addr;
      proxy_set_header    X-Forwarded-for \$remote_addr;
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection "upgrade";

    }
}
```

**Reply by ostrerov** [reply]
I use PostgreSQL, without NGINX or Apache - the problem still remains...

**Comment by ostrerov** [comment]
I have a similar problem. I don't use proxy, I created port-forward to wiki.js app by gateway, and by the way, my file size still 5mb

**Comment by tukusejssirs** [comment]
I experience this issue too with the [`latest`/`2.5.303`](https://github.com/requarks/wiki/pkgs/container/wiki/218180679?tag=latest) Docker image.

I also tried to use the [deprecated method](https://docs.requarks.io/install/config) to configure this, however, that too has no effect.

**Comment by daeqs-one** [comment]
it is not possible to upload files larger than 1 gigabyte

**Comment by gaRt3n** [comment]
This issue still persists that you are not able to change the max file uploads...
Wiki.js 2.5.305

**Comment by pasa2006** [comment]
I have a similar problem with increasing the file upload function. I tried to increase the file upload size to 7MB but it didn't work as others in this online forum mentioned. I tried to set it back to the original 5MB but after saving (apply), it doesn't retain the file upload setting. It goes back to zero (0). Please see the screen clip below. How can I fix this problem? I cannot upload any file to assets.
Thanks,

![Screenshot 2024-12-17 113815](https://github.com/user-attachments/assets/c6bcfaff-9ecc-4258-bfd0-deab08a841ae)

**Comment by Anadema** [comment]
Hello,

Same error with the last version installed yesterday, I change in config.yml restart but the interface keep the last config and the system blok me if I try to upload more than 5 Mo.

 
![image](https://github.com/user-attachments/assets/72c5b416-118c-49f5-8951-95db7a4d890e)

![image](https://github.com/user-attachments/assets/31902d54-de3b-4b5b-a617-fb458352bd65)

**Reply by CWCarlson123** [reply]
We have the same issue, Its getting annoying now having to fight the limitation...3 years...

**Comment by Naiches** [comment]
So, I just spent an entire evening on this issue. I'm using the latest Wiki.js version with Docker. I tried setting this value in the config.yml file and via the UI, neither seemed to work. I noticed the value in the PostgreSQL database was updated, but the UI still displayed: Max 10 files, 5 MB each.

After a while, I removed the custom config and just tried setting it via the UI again. I then tried uploading a file, and it actually worked, even though the UI still said: Max 10 files, 5 MB each.

I looked at the code and noticed that this value is hardcoded in the UI:
https://github.com/Requarks/wiki/blob/a7b9be3bf7fd508a18cf6a6d663092f6cff5a27b/client/components/editor/editor-modal-media.vue#L153

Hopefully, this helps someone in the future. :)

**Comment by SpringWhisper** [comment]
The maximum limit for the number of uploadable files is specified in the code, rather than being dynamically retrieved from the database. 

https://github.com/requarks/wiki/blob/main/client/components/editor/editor-modal-media.vue#L145

The above link indicates the location. If the line number changes, search for "max-files". 

You can change it to a larger value, for example:

max-files='100'

Incidentally, the limit on the number of files on the UI is also written in the code. If you want to change, search for "10 files".
